_Week8/group1/simulate.do

```diff
@@ -0,0 +1,116 @@
+cap program drop salary_simulate // note: did not name program 'simulate' because simulate is a pre-existing Stata command
+
+program define salary_simulate, rclass // define an r-class program so that the program writes into return and we can get stuff out of the program
+	syntax anything
+
+	// clear data in memory
+	clear
+
+	// generate blank data (observations without any content)
+	set obs `anything' // `anything' is number of states sampled from
+	
+	* ---------- generate clusters and associated fixed effects ----------*
+	// generate state cluster
+	generate state = _n // generate state clusters
+	label variable state "State in which college or university is located"
+	
+	generate u_i = rchi2(4)*1000 // generate fixed effects for states
+	generate random1 = runiform() // generate random numbers to be used to vary number of schools per region
+	
+	// generate university clusters within states, with 15-25 universities per state 
+	expand 15 if random1 < 0.3
+	expand 20 if random1 >= 0.3 & random1 < 0.7
+	expand 25 if random1 >= 0.7
+	bysort state: generate university = _n // generate university clusters
+	label variable university "College or university that student attended"
+	
+	generate e_ij = rnormal(0,5)*100 // generate fixed effects for universities
+	generate random2 = runiform()  // generate random numbers to be used to vary number of students per school
+	
+	// generate student observations, with 500-1000 students per university
+	expand 500 if random2 < 0.3
+	expand 750 if random2 >= 0.3 & random2 < 0.8
+	expand 1000 if random2 >= 0.8
+	bysort state university: generate student = _n // generate student variable
+
+	* ---------- generate individual characteristics of students ----------*
+	// generate and label variables for parents education, sat score, and number of AP credits taken in high school
+	generate p1_educ = runiformint(10,18)
+	label variable p1_educ "Student's parent 1's years of educational attainment"
+	generate p2_educ = runiformint(10,18)
+	label variable p2_educ "Student's parent 2's years of educational attainment"
+	generate sat_score = 1200+rnormal(0,70)
+	replace sat_score = trunc(sat_score)
+	label variable sat_score "Student's SAT score"
+	replace sat_score = 1600 if sat_score > 1600
+	replace sat_score = 400 if sat_score < 400
+	generate ap_cred = rpoisson(2)
```

Comment:
Why a Poisson distribution?

Reply:
Good question--not sure we had a justification other than knowing that we wanted 1) it to be an integer and 2) to vary the kinds of distributions we used. We can maybe see if there's a distribution of AP classes available online, I'll save this comment so that we remember to edit accordingly.